src/nemos/regularizer.py

```diff
@@ -621,7 +443,35 @@ def _check_mask(mask: jnp.ndarray):
                 f"Data type {mask.dtype} provided instead!"
             )
 
-    def _get_proximal_operator(
+    def _penalization(
+        self, params: Tuple[DESIGN_INPUT_TYPE, jnp.ndarray]
+    ) -> jnp.ndarray:
+        """
+        Calculate the penalization.
+        """
+        # conform to shape (1, n_features) if param is (n_features,) or (n_neurons, n_features) if
+        # param is (n_features, n_neurons)
+        param_with_extra_axis = jnp.atleast_2d(params[0].T)
+
+        vec_prod = jax.vmap(
+            lambda x: self.mask * x, in_axes=0, out_axes=2
+        )  # this vectorizes the product over the neurons, and adds the neuron axis as the last axis
+
+        masked_param = vec_prod(
+            param_with_extra_axis
+        )  # this masks the param, (group, feature, neuron)
+
+        penalty = jnp.linalg.norm(masked_param, axis=0).sum()
+
+        return penalty
```

Comment:
@BalzaniEdoardo One thing I noticed here is that we are not using the regularization strength and I think that we should be since the proximal operator does